src/sidecar/utils.ts

```diff
@@ -1,7 +1,297 @@
+import * as childProcess from "child_process";
+import fs from "fs";
+import sidecarExecutablePath, { version as currentSidecarVersion } from "ide-sidecar";
+import { normalize } from "path";
+import * as vscode from "vscode";
+import { env, Uri } from "vscode";
+import { EXTENSION_VERSION } from "../constants";
+import { logError } from "../errors";
+import { Logger } from "../logging";
+import { NotificationButtons, showErrorNotificationWithButtons } from "../notifications";
+import { checkSidecarOsAndArch } from "./checkArchitecture";
+import { MOMENTARY_PAUSE_MS, SIDECAR_PORT } from "./constants";
+import { SidecarFatalError } from "./errors";
+import { getSidecarLogfilePath } from "./logging";
+import { SidecarStartupFailureReason } from "./types";
+
+const logger = new Logger("sidecar/utils.ts");
+
+/**
+ * Time to wait after having delivered a SIGTERM to sidecar before
+ * promoting to SIGKILL. The ratio of this to {@link MOMENTARY_PAUSE_MS}
+ * is the number of times {@link killSidecar} will pause+poll loop waiting
+ * for an old (by either version or access token) sidecar to die.
+ **/
+export const WAIT_FOR_SIDECAR_DEATH_MS = 4_000; // 4 seconds.
+
 /**
  * Pause for MOMENTARY_PAUSE_MS.
  */
 export async function pause(delay: number): Promise<void> {
   // pause an iota
   await new Promise((timeout_resolve) => setTimeout(timeout_resolve, delay));
 }
+
```

Comment:
All this moved out from `sidecarManager.ts`. Now it only has the class `SidecarManager`, yay.